source/extensions/filters/common/rbac/engine.h

```diff
@@ -28,9 +30,11 @@ class RoleBasedAccessControlEngine {
    * @param effective_policy_id  it will be filled by the matching policy's ID,
    *                   which is used to identity the source of the allow/deny.
    */
-  virtual bool allowed(const Network::Connection& connection, const Envoy::Http::HeaderMap& headers,
-                       const envoy::api::v2::core::Metadata& metadata,
-                       std::string* effective_policy_id) const PURE;
+  virtual bool
+  allowed(const Network::Connection& connection,
```

Comment:
https://google.github.io/styleguide/cppguide.html#Default_Arguments

> Default arguments are banned on virtual functions, where they don't work properly, and in cases where the specified default might not evaluate to the same value depending on when it was evaluated.

Reply:
I see, thanks for the reference!